Add Inventory Specialist (Senior Cargo) Role

Cprn-Sprkl opened the pull request:
# Description

Adds a time gated role with order approval permissions to logistics.

Since the Inventory Specialist will start with order permissions Cargo Technician will have their access reverted to only start with order permission on low population rounds.

This also updates the bounty computer's access so that anyone with order permissions can skip bounties. Previously it checked specifically for Quartermaster permission.

These additions and changes should improve access control for station funds by removing order permission from the starter cargo role, while adding a role to alleviate the previous problem of needing a Logistics Officer for order permissions. This role will also be expected to teach newer cargo technicians about managing station funds.

---

# TODO

- [x] Place Inventory Specialist spawners on each map
- [x] Update job slots on each map to allow for Inventory Specialist

---

# Changelog

:cl: sprkl
- add: A new senior role for Logistics, the Inventory Specialist.
- tweak: Bounty skips can now be done by anyone with order access.
- tweak: Cargo Technician now only starts with order access on low population rounds.

Cprn-Sprkl committed:
Add Inventory Specialist and Update Orders Access

Adds the Inventory Specialist job, icon, ID, PDA, time tracker, and spawner.

Updates Bounty computer access to allow any role with Orders permission to skip bounties.

Removes Orders from base Cargo Tech access.

Cprn-Sprkl committed:
Final fixes for gear and loadout

Sets up the spawn gear and loadout options for Inventory Specialist.

Cprn-Sprkl committed:
Add InvSpec slots to each map

Allows the InvSpec to be picked on each map. Will probably lead to test issues until the spawner entity can be added to each one as well.

Cprn-Sprkl committed:
Make stamps non-exclusive

Probably won't be an issue since as far as I know stamps can't be equipped into any slots, just held. Still seems good practice though.

Cprn-Sprkl committed:
Inventory Specialist Spawners

Adds the InvSpec spawner entity to the floof maps.

southbridge-fur reviewed (approved):
All map changes are good, just adding a spawner

southbridge-fur commented:
You will need to adjust the entity ID of the spawner for Salturn it looks like

Cprn-Sprkl committed:
Change spawner uid on Saltern

Altered the uid. No other instance of the original numbers were in the file so I'm unsure what the issue was, will just have to see if the new uid works. Map loaded fine on my end before and after this change.

Cprn-Sprkl committed:
Another uid for Saltern

Massively increased the uid number.

Cprn-Sprkl commented:
Getting the same test fail from saltern no matter what number I used for the spawner's uid. Is there something else that could be causing this?

southbridge-fur commented:
> Getting the same test fail from saltern no matter what number I used for the spawner's uid. Is there something else that could be causing this?

I feel like upstream really messed up mapping somewhere because this should be working fine

VividPups commented:
Honestly, we need a training role for salvage. It doesn't make any sense for someone to be on the station side, being an Order taker/crate pusher, to  let me be able to fly a shi,p, go planet side, fight dragons, carps and xenos and mine ore. like every other department has some sort of training role but Salvage doesn't?

Cprn-Sprkl commented:
> Honestly, we need a training role for salvage. It doesn't make any sense for someone to be on the station side, being an Order taker/crate pusher, to let me be able to fly a shi,p, go planet side, fight dragons, carps and xenos and mine ore. like every other department has some sort of training role but Salvage doesn't?

Bring it up in The Logistics Update v1 thread and if there's a bunch of support for it then I can take the ideas from the community and implement it. After whatever is messing with this PR is figured out.

Cprn-Sprkl committed:
Merge remote-tracking branch 'upstream/master' into inventory_specialist_role

southbridge-fur commented:
Alright I'm digging into this to go see what the hell is going on

Cprn-Sprkl committed:
Another Spawner Placement Attempt

Reverted to the base map files, built a release build, then used that for mapping the spawner onto each map. This is the result.

southbridge-fur commented:
I'll be creating a PR after this to relink all these devices that got unlinked randomly.
